### PR TITLE
[autoscaler] Avoid race in no-updaters logic

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

Follow up to https://github.com/ray-project/ray/pull/17194
Mark newly up-to-date nodes active when not using updaters, mirroring current completed update logic. Avoids termination of newly created nodes.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/autoscaler.py [hunk 1]
```diff
@@ -681,6 +681,12 @@ def terminate_unhealthy_nodes(self, nodes: List[NodeID], now: float):
             # nodes with pending or failed status:
             if not node_status == STATUS_UP_TO_DATE:
                 continue
+            # This node is up-to-date. If it hasn't had the chance to produce
+            # a heartbeat, fake the heartbeat now (cf. logic for completed node
+            # updaters).
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/autoscaler.py:686]: what does `cf` mean?

Comment (DmitriGekhtman) [reply python/ray/autoscaler/_private/autoscaler.py:686]: latin for compare. getting a little too academic up in here -- will switch to "see" cuz it's the same number of characters :D

**Review by ijrsvt — APPROVED** [review]
LGTM!

**Review by mwtian — APPROVED** [review]
Hopefully there is a way to `mark_active()` immediately after a node "becomes ready".

**Comment by DmitriGekhtman** [comment]
> Hopefully there is a way to `mark_active()` immediately after a node "becomes ready".

(a) Yeah this marking active business ridiculous -- completing a heartbeat should be a precondition for being considered ready.
(b) For K8s, we're planning to remove all of this heartbeat accounting by implementing a health check for Ray worker pods -- https://github.com/ray-project/ray/issues/17209

**Comment by DmitriGekhtman** [comment]
green, merging

**Pull request merged**